11_FFTBloom/main.cpp

```diff
@@ -0,0 +1,1192 @@
+			struct SConvertParams : CAssetConverter::SConvertParams
+			{
+				virtual inline uint32_t getFinalOwnerQueueFamily(const IGPUImage* image, const core::blake3_hash_t& createdFrom, const uint8_t mipLevel)
+				{
+					return computeFamilyIndex;
+				}
+
+				uint32_t computeFamilyIndex;
+			};
+			SConvertParams params = {};
+			params.computeFamilyIndex = queueFamilyIndex;
+			params.transfer = &m_intendedSubmit;
+			params.utilities = m_utils.get();
+			auto result = reservation.convert(params);
+			// block immediately
+			if (result.copy() != IQueue::RESULT::SUCCESS)
+				return false;
+		}
+
+		// Create Out Image
+		{
+			auto dstImgViewInfo = m_srcImageView->getCreationParameters();
+
+			IGPUImage::SCreationParams dstImgInfo(dstImgViewInfo.image->getCreationParameters());
+			// Specify we want this to be a storage image, + transfer for readback (blit when we have swapchain up)
+			dstImgInfo.usage = IImage::EUF_STORAGE_BIT | IImage::EUF_TRANSFER_SRC_BIT;
+			dstImgInfo.format = useHalfFloats ? EF_R16G16B16A16_SFLOAT : EF_R32G32B32A32_SFLOAT;
+			m_outImg = m_device->createImage(std::move(dstImgInfo));
+
+			m_outImg->setObjectDebugName("Convolved Image");
+
+			auto memReqs = m_outImg->getMemoryReqs();
+			memReqs.memoryTypeBits &= m_device->getPhysicalDevice()->getDeviceLocalMemoryTypeBits();
+			auto gpuMem = m_device->allocate(memReqs, m_outImg.get());
+
+			dstImgViewInfo.image = m_outImg;
+			dstImgViewInfo.subUsages = IImage::EUF_STORAGE_BIT | IImage::EUF_TRANSFER_SRC_BIT;
+			dstImgViewInfo.format = useHalfFloats ? EF_R16G16B16A16_SFLOAT : EF_R32G32B32A32_SFLOAT;
+			m_outImgView = m_device->createImageView(IGPUImageView::SCreationParams(dstImgViewInfo));
+
+			m_outImgView->setObjectDebugName("Convolved Image View");
+		}
+
+		// agree on formats
+		const E_FORMAT srcFormat = m_srcImageView->getCreationParameters().format;
+		// TODO: this might be pointless?
+		uint32_t srcNumChannels = getFormatChannelCount(srcFormat);
+		uint32_t kerNumChannels = getFormatChannelCount(m_kerImageView->getCreationParameters().format);
+		//! OVERRIDE (we dont need alpha)
+		srcNumChannels = CHANNELS;
+		kerNumChannels = CHANNELS;
+		assert(srcNumChannels == kerNumChannels); // Just to make sure, because the other case is not handled in this example
+
+		// Compute (kernel) padding size
+
+		// Kernel pixel to image pixel conversion ratio
+		const float bloomRelativeScale = 0.25f;
+		const auto kerDim = m_kerImageView->getCreationParameters().image->getCreationParameters().extent;
+		const auto srcDim = m_srcImageView->getCreationParameters().image->getCreationParameters().extent;
+		bloomScale = core::min(float(srcDim.width) / float(kerDim.width), float(srcDim.height) / float(kerDim.height)) * bloomRelativeScale;
+		if (bloomScale > 1.f)
+			std::cout << "WARNING: Bloom Kernel will Clip and loose sharpness, increase resolution of bloom kernel!" << std::endl;
```

Comment:
ah ok I see where you got it from, you can just `assert` for this example, no one will use it willy nilly with any image (it was important for denoiser since it takes any image as input)